File: ios/DocuPassViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(DocuPassViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(reference, NSString)
RCT_EXPORT_VIEW_PROPERTY(partyId, NSString)
RCT_EXPORT_VIEW_PROPERTY(baseUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(onResult, RCTDirectEventBlock)

@end

File: android/src/main/java/com/idanalyzer/docupass/rn/DocuPassViewManager.kt
package com.idanalyzer.docupass.rn

import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class DocuPassViewManager : SimpleViewManager<DocuPassHostView>() {

    override fun getName() = "DocuPassView"

    override fun createViewInstance(reactContext: ThemedReactContext) = DocuPassHostView(reactContext)

    @ReactProp(name = "reference")
    fun setReference(view: DocuPassHostView, value: String?) {
        view.reference = value
        view.applyProps()
    }

    @ReactProp(name = "partyId")
    fun setPartyId(view: DocuPassHostView, value: String?) {
        view.partyId = value
        view.applyProps()
    }

    @ReactProp(name = "baseUrl")
    fun setBaseUrl(view: DocuPassHostView, value: String?) {
        view.baseUrl = value
        view.applyProps()
    }

    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any> =
        mapOf("onResult" to mapOf("registrationName" to "onResult"))
}
